### PR TITLE
`CircleCI`: changed all jobs to use Xcode 14

### PR DESCRIPTION
**Pull request by NachoSoto**
Now that [Xcode 14 RC](https://discuss.circleci.com/t/xcode-14-rc-released/45334) is out, we should make it the default for all the jobs moving forward.

### CODE REVIEW

**Comment by NachoSoto** [comment]
Let's see if they all pass though 🍿

**Comment by NachoSoto** [comment]
Ugh. Mac building fails because Xcode 14 doesn't include the new SDK, and so checking for swift's version is the wrong check... I'll downgrade that job for now.

File: .circleci/config.yml [hunk 1]
```diff
@@ -561,43 +561,45 @@ workflows:
...
       - run-test-ios-13:
-          xcode_version: '13.4.1'
+          xcode_version: '14.0.0'
+          <<: *release-branches-and-main
       - run-test-ios-12:
-          xcode_version: '13.4.1'
+          xcode_version: '14.0.0'
+          <<: *release-branches-and-main
       - build-tv-watch-and-macos:
           xcode_version: '13.4.1'
```
Comment (tonidero) [inline .circleci/config.yml:582]: Should we add a comment here to revisit it in the future?

**Review by tonidero — APPROVED** [review]
LGTM!

File: .circleci/config.yml (hunk 1 above)
Comment (NachoSoto) [reply .circleci/config.yml:582]: Good idea, added.

**Pull request merged**